packages/app/obojobo-document-engine/src/scripts/oboeditor/components/node/editor-component.js

```diff
@@ -50,6 +50,11 @@ class Node extends React.Component {
 		Transforms.select(this.props.editor, Editor.start(this.props.editor, targetPath))
 	}
 
+	validateId(newId) {
+		const validation = /[^a-z|A-Z|0-9|\-|_|:|.]/
+		return validation.test(newId)
+	}
```

Comment:
Duplicated code - I'm thinking move the validation function into it's own file in util, and import it where needed